src/foundry/client/pixi/layers/controls/ruler.d.mts

```diff
@@ -135,49 +138,126 @@ declare global {
+         * Snap the destination?
          * @defaultValue `true`
          */
-        gridSpaces: boolean;
+        snap: boolean;
+
         /**
-         * Do the measure whatever is the destination point?
+         * If not forced and the destination matches the current destination of this ruler, no measuring is done and nothing is returned
          * @defaultValue `false`
          */
         force: boolean;
       }>,
-    ): Ruler.Segment[];
+    ): Ruler.MeasurementSegment[] | void;
+
+    /**
+     * Get the measurement origin.
+     * @param point   - The waypoint
+     * @param options - Additional options
+     */
+    protected _getMeasurementOrigin(
+      point: Point,
+      options?: InexactPartial<{
+        /**
+         * Snap the waypoint?
+         * @defaultValue `true`
+         */
+        snap: boolean;
+      }>,
+    ): Point;
 
     /**
      * While measurement is in progress, update the destination to be the central point of the target grid space.
-     * @param destination - The current pixel coordinates of the mouse movement
-     * @returns The destination point, a center of a grid space
+     * @param point   - The point coordinates
+     * @param options - Additional options
+     * @returns The snapped destination point
      */
-    protected _getMeasurementDestination(destination: Point): Point;
+    protected _getMeasurementDestination(
+      point: Point,
+      options?: InexactPartial<{
+        /**
+         * Snap the point?
+         * @defaultValue `true`
+         */
+        snap: boolean;
+      }>,
+    ): Point;
 
     /**
      * Translate the waypoints and destination point of the Ruler into an array of Ray segments.
      * @returns The segments of the measured path
      */
-    protected _getMeasurementSegments(): RulerMeasurementSegment[];
+    protected _getMeasurementSegments(): Ruler.MeasurementSegment[];
+
+    /**
+     * Handle the start of a Ruler measurement workflow
+     * @param origin  - The origin
+     * @param options - Additional options
+     */
+    protected _startMeasurement(
+      origin: Point,
+      options?: InexactPartial<{
+        /**
+         * Snap the origin?
+         * @defaultValue `true`
+         */
+        snap: boolean;
+
+        /**
+         * The token that is moved (defaults to {@link Ruler#_getMovementToken})
+         */
+        token: Token.ConfiguredInstance | null;
+      }>,
+    ): void;
+
+    /**
+     * Handle the conclusion of a Ruler measurement workflow
+     */
+    protected _endMeasurement(): void;
+
+    /**
+     * Handle the addition of a new waypoint in the Ruler measurement path
+     * @param point   - The waypoint
+     * @param options - Additional options
+     */
+    protected _addWaypoint(
+      point: Point,
+      options?: InexactPartial<{
+        /**
+         * Snap the waypoint?
+         * @defaultValue `true`
+         */
+        snap: boolean;
+      }>,
```

Comment:
Nit; this is reused enough you could probably make an interface for it.

Reply:
The problem is they change the comment around and it's not always *just* snap

Reply:
I see